homeassistant/components/emoncms/sensor.py

```diff
@@ -107,6 +109,9 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         if sensor_names is not None:
             name = sensor_names.get(int(elem["id"]), None)
 
+        if config.get(CONF_USE_EMONCMS_UNIT):
+            unit_of_measurement = elem.get("unit","")
```

Comment:
Furthermore, if no unit of measurement is found here an empty string is used which is not ideal.
Either set it to `None` or use the unit that has been used before for all sensors.

Reply:
Would using the DEFAULT_UNIT be OK?

Reply:
Yes 👍

Reply:
As above how about
```
unit_of_measurement = elem.get("unit",config.get(CONF_UNIT_OF_MEASUREMENT))
```